DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -117,16 +108,11 @@ struct NumberOfActiveBits<0> {
 };
 
 /// evaluate the array size necessary to hold a N-byte number with type T
-template <int N, typename T>
-struct ArraySize {
-  static_assert(N >= sizeof(T), "get this code to work first");
-  static int const value = 1;
-  //static int const value = std::conditional<(N > sizeof(T)), ArraySize<(N - sizeof(T)), T>, ArraySize<0, T>>::value + 1;
-};
-template <typename T>
-struct ArraySize<0, T> {
-  static int const value = 0;
-};
+template <typename T, int N>
+constexpr int ArraySize() {
+  if (N<=sizeof(T)) return 1;
+  return N/sizeof(T)+(N%sizeof(T));
```

Comment:
also, I think it is OK to not require the multiple of 2 here - it will give you a minimal array size to hold N bytes. The restriction I would only make in the declaration of a type that uses it.